### PR TITLE
code analysis and circleci

### PR DESCRIPTION
**Pull request by leandrodzp**
Description:
* Add code analysis tools
* Add circleCi

### CODE REVIEW

File: .circleci/application.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SERVER_URL: 'localhost:3000'
+SECRET_KEY: 'f8244d15aae4b14006b78cd239f4cce5188f321345f32e6909a20cceb9d5c8d78ecf2089b5fece908423262827334ddd7b7e160c013edb273e48cdfc55564f7f'
```
Comment (letiesperon) [inline .circleci/application.yml]: You shouldn't need this key in the test enviornment

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Lint/DuplicatedKey:
+ Exclude:
+   - spec/spec_helper.rb
+
+Metrics/AbcSize:
+ # The ABC size is a calculated magnitude, so this number can be an Integer or
+ # a Float.
+ Exclude:
+   - app/models/user.rb
```
Comment (letiesperon) [inline .rubocop.yml]: Why are you excluding all these models from the ABC size metrics? Try to make the methods smaller so you don't have to exclude them

Comment (gonzaM) [reply .rubocop.yml]: Oh, I sent him the file to copy the configurations, my fault 🤦‍♂️ , he'll remove it. Thanks Leti!

**Comment by gonzaM** [comment]
Merging since the comments were fixed and he need the PR 👍

**Pull request merged**